catalog/cataloger_export_test.go

```diff
@@ -177,106 +174,39 @@ func TestExportState(t *testing.T) {
 	c := testCataloger(t)
 	repo := testCatalogerRepo(t, ctx, c, prefix, defaultBranch)
 
-	cases := []struct {
-		name         string
-		startRef     string // start with this ref (and state) if set, otherwise start with no row
-		startState   CatalogBranchExportStatus
-		startMessage *string
-		setRef       string
-		expectState  CatalogBranchExportStatus
-		expectErr    func(t *testing.T, err error)
-	}{
-		{
-			name:        "clean",
-			setRef:      ref2,
-			expectState: ExportStatusInProgress,
-			expectErr: func(t *testing.T, err error) {
-				if err != nil {
-					t.Errorf("unexpected error %s", err)
-				}
-			},
-		}, {
-			name:        "reset",
-			startRef:    ref1,
-			setRef:      ref2,
-			expectState: ExportStatusInProgress,
-			expectErr: func(t *testing.T, err error) {
-				if err != nil {
-					t.Errorf("unexpected error %s", err)
-				}
-			},
-		}, {
-			name:        "previousSucceeded",
-			startRef:    ref1,
-			startState:  ExportStatusSuccess,
-			setRef:      ref2,
-			expectState: ExportStatusInProgress,
-			expectErr: func(t *testing.T, err error) {
-				if err != nil {
-					t.Errorf("unexpected error %s", err)
-				}
-			},
-		}, {
-			name:         "previousFailed",
-			startRef:     ref1,
-			startState:   ExportStatusFailed,
-			startMessage: swag.String("humpty dumpty had a great fall"),
-			setRef:       ref2,
-			expectState:  ExportStatusInProgress,
-			expectErr: func(t *testing.T, err error) {
-				if !errors.Is(err, ErrExportFailed) {
-					t.Errorf("expected ErrExportFailed but got %s", err)
-				}
-			},
-		},
+	var insertStart ExportStateCallback
+	insertStart = func(oldRef string, state CatalogBranchExportStatus) (newRef string, newState CatalogBranchExportStatus, newMessage *string, err error) {
+		return ref1, ExportStatusInProgress, nil, nil
 	}
-	for _, tt := range cases {
-		pool, err := pgxpool.Connect(ctx, c.DbConnURI)
-		if err != nil {
-			t.Fatalf(err.Error())
+
+	if err := c.ExportStateSet(repo, defaultBranch, insertStart); err != nil {
+		t.Fatal(err)
+	}
+
+	var InProgressToSuccess ExportStateCallback
```

Comment:
1. Why not define it as a regular function, outside of the test?
2. In any case, why is it capitalized?